src/render/extract.rs

```diff
@@ -75,52 +75,52 @@ pub fn vector_instances(
     }
 }
 
-#[derive(Component, Clone)]
+#[derive(Component)]
 pub struct ExtractedRenderText {
-    pub font: Handle<VelloFont>,
+    pub font: VelloFont,
     pub text: VelloText,
     pub transform: GlobalTransform,
     pub render_mode: CoordinateSpace,
 }
 
-impl ExtractComponent for ExtractedRenderText {
-    type Query = (
-        &'static Handle<VelloFont>,
-        &'static VelloText,
-        &'static GlobalTransform,
-        &'static CoordinateSpace,
-    );
-
-    type Filter = ();
-    type Out = Self;
-
-    fn extract_component(
-        (vello_font_handle, text, transform, render_mode): bevy::ecs::query::QueryItem<
-            '_,
-            Self::Query,
-        >,
-    ) -> Option<Self> {
-        Some(Self {
-            font: vello_font_handle.clone(),
-            text: text.clone(),
-            transform: *transform,
-            render_mode: *render_mode,
-        })
+pub fn text_instances(
+    mut commands: Commands,
+    query_vectors: Extract<
+        Query<(
+            &Handle<VelloFont>,
+            &VelloText,
+            &GlobalTransform,
+            &CoordinateSpace,
+        )>,
+    >,
+    assets: Extract<Res<Assets<VelloFont>>>,
+) {
+    for (vello_font_handle, vello_text, transform, coordinate_space) in
+        query_vectors.iter()
+    {
+        if let Some(asset) = assets.get(vello_font_handle) {
+            commands.spawn(ExtractedRenderText {
+                font: *asset.to_owned(),
```

Comment:
This is where I'm skeptical: "to_owned". We are cloning the asset which is a non-zero cost every frame. Do we have to do this?

Reply:
> This clone happened before in the old extraction way, so nothing got worse efficiency wise

Ditto